src/sensor/ping.cpp

```diff
@@ -47,8 +47,18 @@ Ping::Ping() : Sensor()
         {
             qCWarning(PING_PROTOCOL_PING) << "Can't write in this type of link.";
             _requestTimer.stop();
+            _periodicRequestTimer.stop();
             return;
         }
+
+        if(!link()->isOpen())
+        {
+            qCCritical(PING_PROTOCOL_PING) << "Can't write, port is not open!";
+            _requestTimer.stop();
```

Comment:
Does this line belong here?

Looks like it should go (if not there already) in the _requestTimer callback. While it won't hurt here, it looks like a short piece of spaghetti.

Reply:
This is the requestTimer callback, but I did what you said in requestPeriodicTimer